# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build egl-opengl-stubs-build libpng-build tiff-build"

PKG_CONFIG_PATH=/fb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.13.4; lt-2.4.2)
libtoolize -c -f; aclocal; autoconf; autoheader -f; automake -a -c
PKG_CONFIG_PATH=/fb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/fb/lib ./configure --disable-gl-x11 --disable-glesv1-cm --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install yagears $DESTDIR/fb/bin

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/yagears-build.pc << EOF
Name: yagears
Version: 0.7
Description: Yet Another Gears OpenGL demo
Requires: $REQUIRES

exec=\\
/fb/bin/yagears
EOF

exit
--- yagears-0.7.orig/configure.ac
+++ yagears-0.7/configure.ac
@@ -1,6 +1,7 @@
 AC_INIT(yagears, 0.7)
 
 AC_CONFIG_HEADERS(config.h)
+AC_CONFIG_MACRO_DIR([m4])
 
 AM_INIT_AUTOMAKE
 
@@ -122,6 +123,7 @@
    test x$enable_egl_wayland = xyes || test x$enable_egl_drm; then
   PKG_CHECK_MODULES(EGL, egl, enable_egl=yes, enable_egl=no)
   if test x$enable_egl = xyes; then
+    EGL_CFLAGS="$EGL_CFLAGS -DEGL_NO_PLATFORM_SPECIFIC_TYPES"
     if test x$enable_egl_x11 = xyes; then
       PKG_CHECK_MODULES(X11, x11,
                         AC_DEFINE(EGL_X11, 1, Support for EGL with Xlib platform),
--- yagears-0.7.orig/main.c
+++ yagears-0.7/main.c
@@ -821,7 +821,7 @@
   #if defined(EGL_FBDEV)
   if (!strcmp(backend, "egl-fbdev")) {
     setenv("EGL_PLATFORM", "fbdev", 1);
-    egl_dpy = eglGetDisplay((EGLNativeDisplayType)fb_dpy);
+    egl_dpy = eglGetDisplay((EGLNativeDisplayType)(long)fb_dpy);
   }
   #endif
   #if defined(EGL_WAYLAND)
@@ -974,7 +974,7 @@
   }
   #endif
   #if defined(GL_FBDEV) || defined(EGL_FBDEV)
-  if (!strcmp(backend, "gl-fbdev")) {
+  if (!strcmp(backend, "gl-fbdev") || !strcmp(backend, "egl-fbdev")) {
     fb_win = calloc(1, sizeof(struct fb_window));
     if (!fb_win) {
       printf("fb_window calloc failed\n");
